### PR TITLE
[Cloud_Spanner_to_GCS_Avro] Refactoring InformationSchemaScannerIT

### PR DESCRIPTION
**Pull request by darshan-sj**
The PR refactors the test to use the new test framework to create and utilize the resources and optimizes the InformationSchemaScannerIT integration test suite. By aggressively migrating standalone test cases into a shared, multiplexed Spanner database, and eliminating low-value standalone tests, the suite's overall execution time has been reduced.

It also includes code hygiene improvements and introduces contribution guidelines to ensure the test suite remains performant as it scales.

**Key Changes**
* **Performance Optimizations (Execution Time Reduction):**
Merged tableWithAllTypes into Shared Suite: Migrated tableWithAllTypes from a standalone @Test into a SharedTestCase under sharedInformationSchemaScannerTestGsql. To facilitate this, the FileDescriptorSet (Proto Bundle) construction was moved up to setupSharedResourceManagers(), allowing the shared GSQL database to safely load and test Proto columns.
Deleted Low-Value Isolated Tests: Removed emptyDatabase, pgEmptyDatabase, defaultTimeZone, and pgDefaultTimeZone. These tests added minimal coverage but contributed heavily to the total runtime due to the high overhead of spinning up isolated Spanner databases.

* **Bug Fixes:**
Fixed InformationSchemaScanner Edge Case Bug: Discovered and fixed a critical bug in InformationSchemaScanner.java. When scanning property graphs, the scanner would prematurely execute a return; if a property graph lacked edge tables, terminating the loop. This was changed to continue; to ensure all subsequent property graphs in a shared database are correctly scanned.

* **Code Hygiene & Readability:**
Added Contribution Guidelines: Documented contribution rules in a large block comment at the top of the test class. The guidelines explain how to use SharedTestCase, the shared database context, and required naming conventions (t_<test_case_name>_<suffix>) to prevent future developers from introducing performance regressions.

### CODE REVIEW

File: it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/spanner/SpannerResourceManager.java [hunk 1]
```diff
@@ -334,11 +385,18 @@ public synchronized void executeDdlStatements(List<String> statements)
...
+                DatabaseInfo.Builder dbBuilder =
+                    databaseAdminClient.newDatabaseBuilder(
+                        DatabaseId.of(projectId, instanceId, databaseId));
+                if (protoDescriptors != null) {
+                  dbBuilder.setProtoDescriptors(protoDescriptors);
+                }
+                Database database = dbBuilder.build();
+                databaseAdminClient
+                    .updateDatabaseDdl(database, statements, /* operationId= */ null)
+                    .get();
```
[the remark this answers is not in the document]
Comment (darshan-sj) [reply it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/spanner/SpannerResourceManager.java:400]: Incorrect comment

File: v1/src/main/java/com/google/cloud/teleport/spanner/ddl/InformationSchemaScanner.java [hunk 2]
```diff
@@ -1274,7 +1274,7 @@ private void listPropertyGraphTables(Ddl.Builder builder, String tableType) {
         tablesJson = resultSet.getJson(2);
       } catch (Exception edgeTableException) {
         LOG.debug(propertyGraphNameQualified + " does not contain any edge tables");
-        return;
+        continue;
       }
```
[the remark this answers is not in the document]
Comment (darshan-sj) [reply v1/src/main/java/com/google/cloud/teleport/spanner/ddl/InformationSchemaScanner.java]: Done.

**Review by pratickchokhani — APPROVED** [review]
